source/jormungandr/jormungandr/scenarios/ridesharing/karos.py

```diff
@@ -141,8 +155,9 @@ def _make_response(self, raw_json):
             res.total_seats = None
 
             res.pickup_date_time = offer.get('driverDepartureDate')
-            if res.pickup_date_time is not None:
-                res.dropoff_date_time = res.pickup_date_time + offer.get('duration')
+            res.departure_date_time = res.pickup_date_time - res.origin_pickup_duration
+            res.dropoff_date_time = res.pickup_date_time + offer.get('duration')
```

Comment:
Can we guarantee that res.pickup_date_time exists?

Reply:
yes, pickup_date_time is mandatory in Karos API